### PR TITLE
refactor: indexer

### PR DESCRIPTION
**Pull request by sxzz**
Merge after #21

### CODE REVIEW

File: src/indexer/subapis/stat.ts [hunk 1]
```diff
@@ -3,23 +3,20 @@ import {
...
    */
-  async getStatOfCharacter(
-    characterId: Numberish,
-  ): Promise<CharacterStatEntity> {
-    const url = `${this.endpoint}/stat/characters/${characterId}?`
-
-    const res = await this.fetch(url).then((res) => res.json())
-
-    return res as CharacterStatEntity
+  getByCharacter(characterId: Numberish) {
```
Comment (Songkeys) [inline src/indexer/subapis/stat.ts:17]: Of, or for.

File: src/indexer/subapis/stat.ts [hunk 2]
```diff
@@ -29,14 +26,8 @@ export class StatIndexer extends BaseIndexer {
...
-  async getStatOfNote(
-    characterId: Numberish,
-    noteId: Numberish,
-  ): Promise<NoteStatEntity> {
-    const url = `${this.endpoint}/stat/notes/${characterId}/${noteId}?`
-
-    const res = await this.fetch(url).then((res) => res.json())
-
-    return res as NoteStatEntity
+  getByNote(characterId: Numberish, noteId: Numberish) {
```
Comment (Songkeys) [inline src/indexer/subapis/stat.ts:29]: Of, or for

File: src/indexer/subapis/notification.ts [hunk 3]
```diff
@@ -16,7 +17,7 @@ export class NotificationIndexer extends BaseIndexer {
    * @param options - The options to send to the indexer.
    * @returns The list of notifications.
    */
-  async getNotificationsOfCharacter(
+  getManyByCharacter(
```
Comment (Songkeys) [inline src/indexer/subapis/notification.ts:20]: getMany should be fine as we don't have any alternative.

File: src/indexer/subapis/minted_note.ts [hunk 4]
```diff
@@ -17,7 +18,7 @@ export class MintedNoteIndexer extends BaseIndexer {
    * @param options - The options to send to the indexer.
    * @returns The list of minted notes.
    */
-  async getMintedNotesOfAddress(
+  getManyByAddress(
```
Comment (Songkeys) [inline src/indexer/subapis/minted_note.ts:21]: Of

File: src/indexer/subapis/minted_note.ts [hunk 5]
```diff
@@ -65,7 +64,7 @@ export class MintedNoteIndexer extends BaseIndexer {
    * @param options - The options to send to the indexer.
    * @returns The list of minted notes.
    */
-  async getMintedNotesOfNote(
+  getManyByNote(
```
Comment (Songkeys) [inline src/indexer/subapis/minted_note.ts:67]: Of

**Comment by Songkeys** [comment]
Great. Just some naming issue.

**Review by Songkeys — CHANGES_REQUESTED** [review]

**Comment by sxzz** [comment]
Done

File: src/indexer/subapis/base.ts [hunk 6]
```diff
@@ -43,6 +43,17 @@ export class BaseIndexer {
     return fetch(this.endpoint + url, {
       ...this.fetchOptions,
       ...options,
-    }).then((r) => r.json())
+    }).then(async (r) => {
+      if (r.status !== 200) {
```
Comment (Songkeys) [inline src/indexer/subapis/base.ts:47]: use `res.ok` for all 20X status

**Review by Songkeys — CHANGES_REQUESTED** [review]
One more change.

**Pull request merged**